### PR TITLE
Update to 5.1

### PR DESCRIPTION
**Pull request by seb-exiin**
Various code updates to upgrade UINavigation to UE 5.1

### CODE REVIEW

File: Source/UINavigation/Private/UINavInputContainer.cpp [hunk 1]
```diff
@@ -301,11 +301,9 @@ int UUINavInputContainer::GetOffsetFromTargetColumn(const bool bTop) const
 		case 2:
 			if (bTop) return (TargetColumn == ETargetColumn::Right);
 			else return -(TargetColumn != ETargetColumn::Right);
-			break;
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavInputContainer.cpp:304]: Why did you remove this? This will change the behavior of the switch block...

File: Source/UINavigation/Private/UINavWidget.cpp [hunk 2]
```diff
@@ -801,11 +801,13 @@ void UUINavWidget::RemoveFromParent()
 	Super::RemoveFromParent();
 }
 
+#if ENGINE_MAJOR_VERSION == 5 && ENGINE_MINOR_VERSION < 1
```
Comment (goncasmage1) [inline Source/UINavigation/Private/UINavWidget.cpp:804]: What happened here? Is this no longer available in 5.1?

File: Source/UINavigation/Public/UINavMacros.h [hunk 3]
```diff
@@ -7,4 +7,4 @@
 #define DISPLAYERROR(Text) GEngine->AddOnScreenDebugMessage(-1, 10.f, FColor::Red, FString::Printf(TEXT("%s"), *(FString(TEXT("Error in ")).Append(GetName()).Append(TEXT(": ")).Append(Text))))
 #define DISPLAYERROR_STATIC(Widget, Text) GEngine->AddOnScreenDebugMessage(-1, 10.f, FColor::Red, FString::Printf(TEXT("%s"), *(FString(TEXT("Error in ")).Append(Widget->GetName()).Append(TEXT(": ")).Append(Text))))
 #define DISPLAYWARNING(Text) GEngine->AddOnScreenDebugMessage(-1, 10.f, FColor::Orange, FString::Printf(TEXT("%s"), *(FString(TEXT("Warning in ")).Append(GetName()).Append(TEXT(": ")).Append(Text))))
-#define IS_VR_PLATFORM !PLATFORM_SWITCH && !PLATFORM_XBOXONE
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavMacros.h:10]: Is there no more XBOX_ONE or did it change name?

File: Source/UINavigation/Public/UINavPCComponent.h [hunk 4]
```diff
@@ -133,7 +133,7 @@ class UINAVIGATION_API UUINavPCComponent : public UActorComponent
 
 	void CallCustomInput(const FName ActionName, const bool bPressed);
 
-	void OnControllerConnectionChanged(bool bConnected, FPlatformUserId UserId, int32 UserIndex);
+	virtual void OnControllerConnectionChanged(EInputDeviceConnectionState NewConnectionState, FPlatformUserId PlatformUserId, FInputDeviceId InputDeviceId);
```
Comment (goncasmage1) [inline Source/UINavigation/Public/UINavPCComponent.h:136]: Just noticed, this should probably have an override at the end

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

File: Source/UINavigation/Private/UINavInputContainer.cpp (hunk 1 above)
Comment (seb-exiin) [reply Source/UINavigation/Private/UINavInputContainer.cpp:304]: this is correct, it doesn't change the behavior or Switch. I've removed it as my compiler keep complaining about unreachable code.

`if return and else return`
that means that the compiler will never reach the break as in all the case it returns before

The structure of a switch is:
Switch(v)
case X:
case Y:
etc...

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 2 above)
Comment (seb-exiin) [reply Source/UINavigation/Private/UINavWidget.cpp:804]: Yes for some reason the "OnLevelRemovedFromWorld" isn't on the UWidget anymore

The other solution would be to remove the "virtual" from the function in the .h, but that would simply means the function still exist but is never called by the engine.

File: Source/UINavigation/Public/UINavMacros.h (hunk 3 above)
Comment (seb-exiin) [reply Source/UINavigation/Public/UINavMacros.h:10]: IDK, the compiler complains about the Xbox One but not the Switch...

File: Source/UINavigation/Public/UINavPCComponent.h (hunk 4 above)
Comment (seb-exiin) [reply Source/UINavigation/Public/UINavPCComponent.h:136]: No I made it as a virtual to allow the possibility to override it in a child class of UINavPCComponent.
But if you want me, I can simply remove the Virutal on it!

What do I do?

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 2 above)
Comment (goncasmage1) [reply Source/UINavigation/Private/UINavWidget.cpp:804]: I'd say just remove the function from UINavWidget entirely.
I don't even know what it's doing here in the first place. (I'm not the one who put this here)

File: Source/UINavigation/Public/UINavPCComponent.h (hunk 4 above)
Comment (goncasmage1) [reply Source/UINavigation/Public/UINavPCComponent.h:136]: But isn't this a function from UINavPCReceiver?

**Review by goncasmage1 — CHANGES_REQUESTED** [review]

File: Source/UINavigation/Public/UINavPCComponent.h (hunk 4 above)
Comment (seb-exiin) [reply Source/UINavigation/Public/UINavPCComponent.h:136]: No, it's declared in line 86, with the "PlatformInputMapper.GetOnInputDeviceConnectionChange()"
It's not a native function of Unreal. 

Then in line 317, you execute the portion of code of the Interface "IUINavPCReceiver".

I've removed the virtual, no big deal about it

File: Source/UINavigation/Private/UINavWidget.cpp (hunk 2 above)
Comment (seb-exiin) [reply Source/UINavigation/Private/UINavWidget.cpp]: Done

**Comment by goncasmage1** [comment]
Merge this into the newly created 5.1 branch instead

**Comment by seb-exiin** [comment]
Base changed :)

**Review by goncasmage1 — APPROVED** [review]
